src/window_service.rs

```diff
@@ -356,25 +357,35 @@ mod test {
 
             let t_responder = responder("window_send_test", blob_sockets[0].clone(), r_responder);
             let mut msgs = Vec::new();
-            for v in 0..10 {
-                let i = 9 - v;
-                let b = resp_recycler.allocate();
-                {
-                    let mut w = b.write();
-                    w.set_index(i).unwrap();
-                    w.set_id(me_id).unwrap();
-                    assert_eq!(i, w.get_index().unwrap());
-                    w.meta.size = PACKET_DATA_SIZE;
-                    w.meta.set_addr(&tn.info.contact_info.ncp);
-                }
-                msgs.push(b);
+            let mut recorder = Recorder::new(Hash::default());
```

Comment:
This is too much logic to put into a unit test. Can you pull it out into its own function? Not sure if it belongs in this module or useful enough to have a better home?  And so much going on there it looks like it could use some tests of its own.  Especially the rogue `truncate` that looks to say, "oops, went too far, but no worries, truncate()."